convokit/storage/storageManager.py

```diff
@@ -0,0 +1,214 @@
+                    self.version = test_version
+                    break
+        else:
+            self.version = version
+
+    @property
+    def full_name(self):
+        return make_full_name(self.corpus_id, self.version)
+
+    def __del__(self):
+        if self.storage_type == 'db':
+            self.client.close()
+
+    def purge_all_collections(self):
+        """
+        If this is a db storageManager, remove all data in the database.
+        """
+        for attr, value in self.__dict__.items():
+            if isinstance(value, DBCollectionMapping):
+                warn(f'Dropping collection {value.collection.name} (storage.{attr})')
+                value.collection.drop()
+            if isinstance(value, Database):
+                warn(f'Dropping db {value.name} (storage.{attr})')
+                for collection_name in value.list_collection_names():
+                    warn(f'\tDropping collection {collection_name} (collection in storage.{attr}))')
+                    value.drop_collection(collection_name)
+
+    @staticmethod
+    def default_storage_mode():
+        tmp_mode = os.environ.get('CONVOKIT_STORAGE_MODE')
+        if tmp_mode in ['mem', 'db']:
+            return tmp_mode
+        else:
+            config = read_or_create_config(
+                os.path.expanduser('~/.convokit/config.yml'))
+            return config['default_storage_mode']
+
+    @staticmethod
+    def purge_db():
+        """
+        Remove all data in the default database. 
+        """
+        warn('purging the DB')
+        client = MongoClient()
+        db = client['convokit']
+        for collection_name in db.list_collection_names():
+            warn(
+                f'\tDropping collection {collection_name} (collection in {db.name}))'
+            )
+            db.drop_collection(collection_name)
+        client.close()
+
+    def setup_collections(self, utterance, conversation, speaker,
+                          convokitmeta):
+        """
+        Setup instance variables self._utterances, self._conversations, self._speakers & self._metas.
+        Should be called as storageManager.setup_collections(Utterance, Conversation, Speaker, ConvoKitMeta)
+        after initializing the storageManager:StorageManager.
+        Utterance, Conversation, Speaker, ConvoKitMeta cannot be imported in the convokit.storage module due 
+        to circular imports, so a call to setup_collections is required to configure a StorageManager with 
+        the correct types. 
+        """
+        self._utterances = self.CollectionMapping('utterances',
+                                                  item_type=utterance)
+        self._conversations = self.CollectionMapping('conversations',
+                                                     item_type=conversation)
+        self._speakers = self.CollectionMapping('speakers', item_type=speaker)
+        self._metas = self.CollectionMapping('metas', item_type=convokitmeta)
```

Comment:
Is there a reason we're making these variables 'private'? Looking at other parts of the code, it looks like the primary way these CollectionMappings are interacted with are by accessing these variables directly, so we should signify that accordingly by doing away with the leading underscore.